tsdb/executor_test.go

```diff
@@ -432,6 +432,82 @@ func TestWritePointsAndExecuteTwoShardsTagSetOrdering(t *testing.T) {
 	}
 }
 
+// Test to ensure the engine handles measurements across stores.
+func TestWritePointsAndExecuteTwoShardsShowMeasurements(t *testing.T) {
+	// Create two distinct stores, ensuring shard mappers will shard nothing.
+	store0 := testStore()
+	defer os.RemoveAll(store0.Path())
+	store1 := testStore()
+	defer os.RemoveAll(store1.Path())
+
+	// Create a shard in each store.
+	database := "foo"
+	retentionPolicy := "bar"
+	store0.CreateShard(database, retentionPolicy, sID0)
+	store1.CreateShard(database, retentionPolicy, sID1)
+
+	// Write two points across shards.
+	pt1time := time.Unix(1, 0).UTC()
+	if err := store0.WriteToShard(sID0, []tsdb.Point{tsdb.NewPoint(
+		"cpu",
+		map[string]string{"host": "serverA"},
+		map[string]interface{}{"value1": 100},
+		pt1time,
+	)}); err != nil {
+		t.Fatalf(err.Error())
+	}
+	pt2time := time.Unix(2, 0).UTC()
+	if err := store1.WriteToShard(sID1, []tsdb.Point{tsdb.NewPoint(
+		"mem",
+		map[string]string{"host": "serverB"},
+		map[string]interface{}{"value2": 200},
+		pt2time,
+	)}); err != nil {
+		t.Fatalf(err.Error())
+	}
+	var tests = []struct {
+		skip      bool   // Skip test
+		stmt      string // Query statement
+		chunkSize int    // Chunk size for driving the executor
+		expected  string // Expected results, rendered as a string
+	}{
+		{
+			stmt:     `SHOW MEASUREMENTS`,
+			expected: `[{"name":"measurements","columns":["name"],"values":[["cpu"],["mem"]]}]`,
+		},
+		{
+			stmt:     `SHOW MEASUREMENTS WHERE host='serverB'`,
```

Comment:
Yeah, just to show nothing blows up when no measurements are returned from anywhere in the cluster. Unless you have that covered.